lib/ansible/plugins/inventory/vmware.py

```
@@ -0,0 +1,238 @@
+            raise AnsibleError("Unable to connect to vCenter or ESXi API at %s on TCP/%s: %s" % (self.hostname, self.port, e))
+        except vmodl.fault.InvalidRequest as e:
+            # Request is malformed
+            raise AnsibleError("Failed to get a response from server %s:%s as "
+                               "request is malformed: %s" % (self.hostname, self.port, e.msg))
+        except Exception as e:
+            raise AnsibleError("Unknown error while connecting to vCenter or ESXi API at %s:%s : %s" % (self.hostname, self.port, e))
+
+        if service_instance is None:
+            raise AnsibleError("Unknown error while connecting to vCenter or ESXi API at %s:%s" % (self.hostname, self.port))
+
+        atexit.register(connect.Disconnect, service_instance)
+        return service_instance.RetrieveContent()
+
+    def verify_file(self, path):
+        """
+        Function to verify plugin configuration file and mark this plugin active
+        Args:
+            path: Path of configuration YAML file
+
+        Returns: True if everything is correct, else False
+
+        """
+        valid = False
+        if super(InventoryModule, self).verify_file(path):
+            if path.endswith(('vmware.yaml', 'vmware.yml')):
+                valid = True
+        if not HAS_REQUESTS or not HAS_PYVMOMI:
+            valid = False
+        return valid
+
+    def parse(self, inventory, loader, path, cache=True):
+        """
+        parses the inventory file
+        """
+
+        super(InventoryModule, self).parse(inventory, loader, path, cache=cache)
+
+        # Get Cache key
+        cache_key = self.get_cache_key(path)
+
+        config_data = self._read_config_data(path)
+
+        # set _options from config data
+        self._consume_options(config_data)
+
+        source_data = None
+        if cache:
+            cache = self._options.get('cache')
+
+        self._set_params()
+        self.content = self.login()
```

login returns ansibleerror, should really return a parsing error for inventory manager to skip this source

---

OK.